packages/cashier/src/containers/cashier/cashier.tsx

```diff
@@ -110,8 +111,26 @@ const Cashier = observer(({ history, location, routes: routes_config }: TCashier
 
     React.useEffect(() => {
         if (is_p2p_enabled_success && !is_p2p_enabled && history.location.pathname.startsWith(routes.cashier_p2p)) {
+            const url_params = new URLSearchParams(history.location.search);
+            const advert_id = url_params.get('advert_id');
+
             history.push(routes.cashier_deposit);
+
+            if (advert_id) {
+                if (is_demo) {
+                    toggleReadyToDepositModal();
+                } else {
+                    error.setErrorMessage({
+                        code: 'ShareMyAdsError',
+                        message:
+                            currency !== 'USD' && is_svg
+                                ? localize('Deriv P2P is currently unavailable in this currency.')
+                                : localize('Deriv P2P is currently unavailable in your country.'),
+                    });
+                }
+            }
         }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
handled all the edge cases here as this already checks if p2p is not enabled and includes the routing check if it is p2p, just added one more if to check if advert_id is within the search params

Reply:
so is_svg is not required in the dependency list?

Reply:
so for the is_svg check, i needed to only check the scenario if the currency is not USD and within an svg country as i needed to show the first error message in that scenario, if the user is from a non svg country, we prioritise the second message the be shown since they aren't from an svg country